### PR TITLE
commented exports in script, replaced with defaults in rust in conditionals

### PR DESCRIPTION
**Pull request by debjit-bw**
Here I tried to implement the exports in dev_fed. For the variable vars, I used the defaults for now. #2101

### CODE REVIEW

**Comment by Maan2003** [comment]
just dumping some design ideas:

```rust
// returns error on invalid utf8, Ok(None) on var=="" or var is not set. Ok(Some(value)) if it is set
fn env_var(var: String) -> Result<Option<String>>;
```
```rust
// macro to simplifiy creation of envs
macro_rules! env { ... } 

// example:
env!(MainEnv {
   FOO = env_var("FOO")?.unwrap_or("default_value"),
});


// expands to

struct MainEnv {
    FOO: String,
}

impl MainEnv {
    fn init() -> Result<Self> {
         Ok(Self { FOO: env_var("FOO")?.unwrap_or("default_value"), });
    }

   fn set_global(&self) -> Result<()> {
        env::set_var("FOO", self.FOO)?;
        Ok(())
   }
}
```

**Comment by debjit-bw** [comment]
in fields like `FM_TEST_DIR = "/tmp/fm-XXXXX"`, I couldn't set variables dynamically like `FM_TEST_DIR = &test_dir`. It was saying `can't capture dynamic environment in a fn item; use the '|| { ... }' closure form instead` but using `|| test_dir` wasn't working either.

File: fedimint-bin-tests/src/main.rs [hunk 1]
```diff
@@ -527,7 +527,237 @@ struct DevFed {
...
+macro_rules! env {
+    ($name:ident { $( $var:ident = $value:expr ),* $(,)? }) => {
+        struct $name {
+            $( $var: String, )*
+        }
+
+        impl $name {
+            fn init() -> Result<Self, std::env::VarError> {
+                Ok(Self {
+                    $( $var: env_var(stringify!($var).to_owned())?.unwrap_or_else(|| $value.to_owned()), )*
```
Comment (Maan2003) [inline fedimint-bin-tests/src/main.rs:549]: bind these variables to local variables, so you can use previously defined variables here

Comment (Maan2003) [inline fedimint-bin-tests/src/main.rs:549]: I think this should just be $value and `env_var(....).unwrap_or(..)` should be at call site.

**Pull request closed without merging**